pkg/validation/user.go

```diff
@@ -85,3 +88,56 @@ func ValidateUserUpdate(oldUser, newUser *kubermaticv1.User) field.ErrorList {
 
 	return allErrs
 }
+
+func ValidateUserDelete(ctx context.Context,
+	user *kubermaticv1.User,
+	client ctrlruntimeclient.Client,
+	seedClientGetter provider.SeedClientGetter,
+	namespace string) error {
+	projects, err := kubermaticv1helper.GetUserOwnedProjects(ctx, client, user.Name)
+	if err != nil {
+		return err
+	}
+
+	for _, project := range projects {
+		// if project has multiple owner users
+		if len(project.OwnerReferences) > 1 {
+			continue
+		}
+
+		// project has single owner user then check if project has resources
+
+		// if project has externalclusters
+		hasExtClusters, err := kubermaticv1helper.HasExternalClusters(ctx, client, project.Name)
+		if err != nil {
+			return err
+		}
+		if hasExtClusters {
+			return fmt.Errorf("operation not permitted!: user project %s has resources, clusters or externalclusters", project.Name)
+		}
+
+		// if project has clusters on any seed
+		listOpts := &ctrlruntimeclient.ListOptions{
+			Namespace: namespace,
+		}
+		seeds := &kubermaticv1.SeedList{}
+		if err := client.List(ctx, seeds, listOpts); err != nil {
+			return fmt.Errorf("failed to list the seeds: %w", err)
+		}
+		for _, seed := range seeds.Items {
```

Comment:
Please use a seedsGetter instead. Same place where you got your seedClientGetter from.